app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryModel.kt

```diff
@@ -1,17 +1,21 @@
 package org.simple.clinic.drugs.selection.custom
 
+import org.simple.clinic.drugs.search.Drug
 import org.simple.clinic.drugs.search.DrugFrequency
 
 data class CustomDrugEntryModel(
+    val drug: Drug?,
```

Comment:
Because when in `CREATE` mode we open the bottom sheet from the `DrugSearchScreen` where two cases are possible..
1. The user has typed a drug name which is not part of the drug list. 
2. The user has typed a drug and finds in the drug list and then clicks on it.

So in case 1 we send only the drug name to the model and in case 2 we send the drug object. Later on while initialising the screen we load the drugname, dosage, and frequency based on the drug object or we just load the drug name.

Reply:
Ok, so the two are mutually exclusive. In which case, we should make this explicit by making this part of the `OpenAs` types instead of having all of these at the same level in the model.

Reply:
@janhavisinghh Can you take a look at this as well, please?

Reply:
makes sense, making the change